ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Query.hs

```diff
@@ -751,6 +752,10 @@ querySupportedVersion = \case
     v11 = ShelleyNodeToClientVersion11
     v12 = ShelleyNodeToClientVersion12
 
+querySupportedVersions :: BlockQuery (ShelleyBlock proto era) result -> [ShelleyNodeToClientVersion]
+querySupportedVersions q =
+  [ v | v <- [minBound..maxBound], queryIsSupportedOnVersion q v ]
```

Comment:
Does this function (and the other variant) cross the [fairbairn threshold](https://wiki.haskell.org/Fairbairn_threshold)?